solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudSolrClient.java

```diff
@@ -304,7 +304,8 @@ public ClusterState getClusterState() {
   @Override
   public void close() throws IOException {
     if (this.threadPool != null && !this.threadPool.isShutdown()) {
-      this.threadPool.shutdown();
+      ExecutorUtil.shutdownAndAwaitTermination(this.threadPool);
+      this.threadPool = null;
```

Comment:
As far as I can tell the test ended quickly and the thread pool still hadn't terminated the threads yet. This will ensure this waits until the threadpool is actually shutdown.